inlong-audit/audit-service/src/main/java/org/apache/inlong/audit/cache/RealTimeQuery.java

```diff
@@ -0,0 +1,312 @@
+                Configuration.getInstance().get(KEY_PREP_STMT_CACHE_SQL_LIMIT, DEFAULT_PREP_STMT_CACHE_SQL_LIMIT));
+        config.setMaximumPoolSize(
+                Configuration.getInstance().get(KEY_DATASOURCE_POOL_SIZE,
+                        DEFAULT_DATASOURCE_POOL_SIZE));
+        return new HikariDataSource(config);
+    }
+
+    /**
+     * Query the audit data of log time.
+     *
+     * @param startTime
+     * @param endTime
+     * @param inlongGroupId
+     * @param inlongStreamId
+     * @param auditId
+     * @return
+     */
+    public List<StatData> queryLogTs(String startTime, String endTime, String inlongGroupId,
+            String inlongStreamId, String auditId) {
+        List<StatData> statDataList = new LinkedList<>();
+        for (DataSource dataSource : dataSourceList) {
+            statDataList =
+                    doQueryLogTs(dataSource, startTime, endTime, inlongGroupId, inlongStreamId, auditId);
+            if (!statDataList.isEmpty()) {
+                break;
+            }
+            LOGGER.info("Change another audit source to query data! Params is: {} {} {} {} {}",
+                    startTime, endTime, inlongGroupId, inlongStreamId, auditId);
+        }
+        return statDataList;
+    }
+
+    /**
+     * Do query the audit data of log time.
+     *
+     * @param dataSource
+     * @param startTime
+     * @param endTime
+     * @param inlongGroupId
+     * @param inlongStreamId
+     * @param auditId
+     * @return
+     */
+    private List<StatData> doQueryLogTs(DataSource dataSource, String startTime, String endTime, String inlongGroupId,
+            String inlongStreamId, String auditId) {
+        List<StatData> result = new LinkedList<>();
+        try (Connection connection = dataSource.getConnection();
+                PreparedStatement pstat = connection.prepareStatement(queryLogTsSql)) {
+            pstat.setString(1, startTime);
+            pstat.setString(2, endTime);
+            pstat.setString(3, inlongGroupId);
+            pstat.setString(4, inlongStreamId);
+            pstat.setString(5, auditId);
+            try (ResultSet resultSet = pstat.executeQuery()) {
+                while (resultSet.next()) {
+                    StatData data = new StatData();
+                    data.setLogTs(resultSet.getString(1));
+                    data.setInlongGroupId(resultSet.getString(2));
+                    data.setInlongStreamId(resultSet.getString(3));
+                    data.setAuditId(resultSet.getString(4));
+                    data.setAuditTag(resultSet.getString(5));
+                    data.setCount(resultSet.getLong(6));
+                    data.setSize(resultSet.getLong(7));
+                    data.setDelay(resultSet.getLong(8));
+                    result.add(data);
+                }
+            } catch (SQLException sqlException) {
+                LOGGER.error("Query has SQL exception! ", sqlException);
```

Comment:
Please log the specified data source which throws the exception.

Reply:
Fixed